keyboards/preonic/keymaps/brauner/keymap.c

```diff
@@ -0,0 +1,448 @@
+        tap_code16(keycode);
+        return false;
+    }
+    return true;
+}
+
+static inline bool toggle_layer(enum preonic_layers layer, keyrecord_t *record) {
+    if (record->event.pressed) {
+        layer_on(layer);
+    } else {
+        layer_off(layer);
+    }
+    return false;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+        case LOWER:
+	    return toggle_layer(_LOWER, record);
+        case RAISE:
+	    return toggle_layer(_RAISE, record);
+        case MANAGE:
+	    return toggle_layer(_MANAGE, record);
+        case COMPOSITOR:
+	    return toggle_layer(_COMPOSITOR, record);
+        case LT_CUT_X:
+	    return base_layer_hold_key(C(KC_X), record); /* cut on hold */
+        case LT_COPY_Y:
+	    return base_layer_hold_key(C(KC_C), record); /* copy on hold */
+        case LT_PASTE_P:
+	    return base_layer_hold_key(C(KC_V), record); /* paste on hold */
+        case LT_AE_A:
+	    return base_layer_hold_key(RALT(KC_Q), record); /* 'ä' on hold */
+        case LT_OE_O:
+	    return base_layer_hold_key(RALT(KC_P), record); /* 'ö' on hold */
+        case LT_UE_U:
+	    return base_layer_hold_key(RALT(KC_Y), record); /* 'ü' on hold */
+        case LT_SZ_S:
+	    return base_layer_hold_key(RALT(KC_S), record); /* 'ß' on hold */
+    }
+
+    return true;
+};
+
+bool     muse_mode      = false;
+uint8_t  last_muse_note = 0;
+uint16_t muse_counter   = 0;
+uint8_t  muse_offset    = 70;
+uint16_t muse_tempo     = 50;
+
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (muse_mode) {
+        if (IS_LAYER_ON(_RAISE)) {
+            if (clockwise) {
+                muse_offset++;
+            } else {
+                muse_offset--;
+            }
+        } else {
+            if (clockwise) {
+                muse_tempo += 1;
+            } else {
+                muse_tempo -= 1;
+            }
+        }
+    } else {
+        if (clockwise) {
+            register_code(KC_PGDN);
+            unregister_code(KC_PGDN);
+        } else {
+            register_code(KC_PGUP);
+            unregister_code(KC_PGUP);
+        }
+    }
+    return true;
+}
+
+bool dip_switch_update_user(uint8_t index, bool active) {
+    switch (index) {
+        case 0:
+            if (active) {
+                layer_on(_MANAGE);
+            } else {
+                layer_off(_MANAGE);
+            }
+            break;
+        case 1:
+            if (active) {
+                muse_mode = true;
+            } else {
+                muse_mode = false;
+            }
+    }
+    return true;
+}
+
+LEADER_EXTERNS();
+
+static inline void register_ctrl_sequence(uint16_t keycode) {
+    register_code(KC_RCTL);
+    register_code(keycode);
+    unregister_code(keycode);
+    unregister_code(KC_RCTL);
+}
```

Comment:
You could replace this with `tap_code16(RCTL(keycode))`

Reply:
done